File: SalesOrderManagement/SalesOrder.Model/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace SalesOrder.Model
{
    public class Order
    {
        public virtual int Id { get; set; }

        public virtual DateTime OrderDate { get; set; }

        public virtual Customer Customer { get; set; }

        public virtual Address ShippingAddress { get; set; }

        public virtual Address BillingAddress { get; set; }

        public virtual List<OrderLine> Details { get; set; } = new List<OrderLine>();

        public virtual int TotalValue { get; set; }
    }
}

File: SalesOrder/SalesOrder.Model.VB/Order.vb
Namespace SalesOrder.Model

    Public Class Order

        Public Overridable Property Id As Integer

        Public Overridable Property OrderDate As DateTime

        Public Overridable Property Customer As Customer

        Public Overridable Property ShippingAddress As Address

        Public Overridable Property BillingAddress As Address

        Public Overridable Property Details As ICollection(Of OrderLine) = New List(Of OrderLine)()

        Public Overridable Property TotalValue As Integer
    End Class
End Namespace